pkg/analytics/analytics.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analytics
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"go.buf.build/protocolbuffers/go/prometheus/prometheus"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+type AnalyticsSender struct {
+	logger log.Logger
+
+	client *Client
+
+	machineID string
+	arch      string
+	cpuCores  float64
+	version   string
+}
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+func NewSender(
+	logger log.Logger,
+	client *Client,
+	arch string,
+	cpuCores int,
+	version string,
+) *AnalyticsSender {
+	return &AnalyticsSender{
+		logger:    logger,
+		client:    client,
+		machineID: randSeq(20),
```

Comment:
A new machine id is generated every single time the agent boots up, so we can't tell how many unique machines are running the Agent nor what machine is sending what data, which I assumed was the goal.

Perhaps we could submit the hash of the hostname?